### PR TITLE
Feature/uniffi credential import

### PR DESCRIPTION
**Pull request by zaneschepke**
Add base58 and file credential import to uniffi

### CODE REVIEW

File: nym-vpn-lib/src/platform/mod.rs [hunk 1]
```diff
@@ -149,6 +152,37 @@ pub fn runVPN(config: VPNConfig) -> Result<(), FFIError> {
...
+}
+
+#[allow(non_snake_case)]
+#[uniffi::export]
+pub fn checkCredential(credential: String) -> Result<(), FFIError> {
+    RUNTIME.block_on(check_credential_string(&credential))
+}
+
+async fn check_credential_string(credential: &str) -> Result<(), FFIError> {
+    match check_credential_base58(credential).await {
```
Comment (octol) [inline nym-vpn-lib/src/platform/mod.rs:180]: tip: you can also do this as
```rust
async fn check_credential_string(credential: &str) -> Result<(), FFIError> {
    check_credential_base58(credential).map_err(|_| FFIError::InvalidCredential)
}
```

File: nym-vpn-lib/uniffi/nym_vpn_lib.kt [hunk 2]
```diff
@@ -695,6 +695,10 @@ internal interface UniffiForeignFutureCompleteVoid : com.sun.jna.Callback {
 
 
 
+
+
+
+
```
Comment (octol) [inline nym-vpn-lib/uniffi/nym_vpn_lib.kt:701]: Lots and lots of blank lines?

File: nym-vpn-lib/src/platform/mod.rs [hunk 3]
```diff
@@ -149,6 +152,37 @@ pub fn runVPN(config: VPNConfig) -> Result<(), FFIError> {
...
+async fn import_credential_from_string(credential: &str, path: &str) -> Result<(), FFIError> {
+    let path_result = PathBuf::from_str(path);
+    let path_buf = match path_result {
+        Ok(p) => p,
+        Err(_) => return Err(FFIError::InvalidPath),
+    };
+    match import_credential_base58(credential, path_buf).await {
+        Ok(_) => Ok(()),
+        Err(_) => Err(FFIError::InvalidCredential),
+    }
```
Comment (octol) [inline nym-vpn-lib/src/platform/mod.rs:170]: tip: see other comment about `check_credential_string`

File: nym-vpn-lib/src/platform/mod.rs [hunk 4]
```diff
@@ -149,6 +152,37 @@ pub fn runVPN(config: VPNConfig) -> Result<(), FFIError> {
...
+pub fn importCredential(credential: String, path: String) -> Result<(), FFIError> {
+    RUNTIME.block_on(import_credential_from_string(&credential, &path))
+}
+
+async fn import_credential_from_string(credential: &str, path: &str) -> Result<(), FFIError> {
+    let path_result = PathBuf::from_str(path);
+    let path_buf = match path_result {
+        Ok(p) => p,
+        Err(_) => return Err(FFIError::InvalidPath),
+    };
```
Comment (octol) [inline nym-vpn-lib/src/platform/mod.rs:166]: tip: idiomatic here is

```rust
let path_buf = path_result.map_err(|_| FFIError::InvalidPath)?
```

**Review by octol — APPROVED** [review]

File: nym-vpn-lib/uniffi/nym_vpn_lib.kt (hunk 2 above)
Comment (zaneschepke) [reply nym-vpn-lib/uniffi/nym_vpn_lib.kt:701]: (generated) 😆

**Pull request merged**